vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/index/VXQueryIndexReader.java

```diff
@@ -191,8 +262,13 @@ public void parse(ArrayBackedValueStorage abvsFileNode) throws IOException {
     }
 
     private int buildElement(ArrayBackedValueStorage abvsFileNode, int fieldNum) throws SAXException {
+        List<String> names = new ArrayList<>();
```

Comment:
Move these arrays into IndexAtributes. Then create a single IndexAttributes object that you call reset before using.